# <font color = red > _Daelyn Oliveira_ <font>

## <font color = blue > _Module 2 Report<font> -<font color = blue> Jobs_ <font>

## 1. Objective of the Proposed Research <br>

According to Business Insider the cost of living in San Francisco is 62.6% higher than the US average. The main factors that affect the cost of living are income and housing. An average monthly salary for a family of 4 to live comfortably in San Francisco is around $12,370.  <br>

The San Francisco Chronicle newspaper is conducting an analysis of cost of living in San Francisco city. A database of the salary and benefits paid to City employees is maintained by The San Francisco Controller's Office since fiscal year 2013. We will analyse this data available and the result will be published as a main page artilce next month. <br>

We will analyse the different salaries given to different departments, job types and also the health and dental insurance plans whether covered or not, other additional payments given to employees, benefits and also overtimes of the different job types. 


## 2. Data Exploration

We first import the data in csv format and also strings aren't converted to factors. 

In [1]:
jobs <- read.csv("employee-compensation.csv", stringsAsFactors = F)

The below code shows the first 5 rows of the dataset. 

In [2]:
head(jobs, n=5)

Year.Type Year Organization.Group.Code Organization.Group           
1 Calendar  2028 7                       General City Responsibilities
2 Calendar  2028 1                       Public Protection            
3 Fiscal    2028 1                       Public Protection            
4 Calendar  2019 7                       General City Responsibilities
5 Calendar  2019 1                       Public Protection            
  Department.Code Department Union.Code Union                         
1 229259                     792        Utd Pub EmpL790 SEIU-Crt Clrks
2 CRT                        792        Utd Pub EmpL790 SEIU-Crt Clrks
3 CRT                        792        Utd Pub EmpL790 SEIU-Crt Clrks
4 232108                     911        POA                           
5 DAT                        311        Municipal Attorneys Assoc     
  Job.Family.Code Job.Family      ... Employee.Identifier Salaries  Overtime
1 0000            Untitled        ... 8540990                674.28      0.0
2 0000            Untitled        ... 8540990                674.28      0.0
3 0000            Untitled        ... 8540990                674.28      0.0
4 Q000            Police Services ... 8577148             124709.01 100499.6
5 8100            Legal & Court   ... 8603109             155489.01      0.0
  Other.Salaries Total.Salary Retirement Health.and.Dental Other.Benefits
1    5.76           680.04      130.91       0.00             53.86      
2    5.76           680.04      130.91       0.00             53.86      
3    5.76           680.04      130.91       0.00             53.86      
4 5501.78        230710.41    23271.86   14293.60           3934.00      
5 1500.00        156989.01    29239.75   14308.46          11100.60      
  Total.Benefits Total.Compensation
1   184.77          864.81         
2   184.77          864.81         
3   184.77          864.81         
4 55975.56       286685.97         
5 69326.83       226315.84

The below code shows the bottom 5 rows of the dataset.

In [3]:
tail(jobs, n=5)

Year.Type Year Organization.Group.Code
835303 Fiscal    2013 6                      
835304 Fiscal    2013 5                      
835305 Fiscal    2013 2                      
835306 Fiscal    2013 1                      
835307 Fiscal    2013 2                      
       Organization.Group                      Department.Code
835303 General Administration & Finance        ASR            
835304 Culture & Recreation                    REC            
835305 Public Works, Transportation & Commerce MTA            
835306 Public Protection                       POL            
835307 Public Works, Transportation & Commerce DBI            
       Department                    Union.Code
835303 ASR Assessor / Recorder       790       
835304 REC Recreation & Park Commsn  790       
835305 MTA Municipal Transprtn Agncy   6       
835306 POL Police                    911       
835307 DBI Building Inspection        38       
       Union                              Job.Family.Code
835303 SEIU - Miscellaneous, Local 1021   4200           
835304 SEIU - Miscellaneous, Local 1021   2700           
835305 Electrical Workers, Local 6        7300           
835306 Police Officers' Association       Q000           
835307 Plumbers and Pipefitters, Local 38 6200           
       Job.Family               ... Employee.Identifier Salaries  Overtime
835303 Appraisal & Taxation     ... 48952                36053.00 263.48  
835304 Housekeeping & Laundry   ... 54963                53192.12   0.00  
835305 Journeyman Trade         ... 31177                65876.00 997.34  
835306 Police Services          ... 21887               123487.27   0.00  
835307 Public Safety Inspection ... 40949               105742.01 315.20  
       Other.Salaries Total.Salary Retirement Health.and.Dental Other.Benefits
835303   372.00        36688.48     6633.07    7631.83          3001.15       
835304   467.16        53659.28     9771.30   12851.02          4482.69       
835305 23411.95        90285.29    13015.46   10586.09          7360.81       
835306  1060.04       124547.31    21999.44   11456.99          2113.86       
835307  5288.35       111345.56    19663.52   12801.79          9283.02       
       Total.Benefits Total.Compensation
835303 17266.05        53954.53         
835304 27105.01        80764.29         
835305 30962.36       121247.65         
835306 35570.29       160117.60         
835307 41748.33       153093.89

Let's see how many rows this dataset contains.

In [4]:
nrow(jobs)

[1] 835307

Let's see how many columns this dataset contains.

In [5]:
ncol(jobs)

[1] 22

## 3. Subsets of Data

### Subset 1

Here we create a subset of the mentioned columns having salaries less than 0, employees that do not get paid on time or do not get paid at all.  

In [2]:
s1 <- subset(jobs, Salaries < 0, select=c("Year.Type", "Year", "Organization.Group", "Union", "Job", "Salaries"))

Below we diplay the top 5 rows and also the number of rows this subset has.

In [3]:
head(s1)
nrow(s1)

Year.Type Year Organization.Group Union                         
195   Calendar  2019 Public Protection  Firefighters,Local 798, Unit 1
3921  Calendar  2019 Community Health   SEIU, Local 1021, RN          
12067 Calendar  2019 Community Health   SEIU, Local 1021, RN          
15628 Calendar  2019 Public Protection  Firefighters,Local 798, Unit 1
22067 Calendar  2019 Community Health   SEIU, Local 1021, RN          
25627 Calendar  2019 Public Protection  Firefighters,Local 798, Unit 1
      Job                         Salaries
195   Captain, Emergency Med Svcs -9942.33
3921  Special Nurse                 -10.74
12067 Special Nurse                 -10.74
15628 Captain, Emergency Med Svcs -9942.33
22067 Special Nurse                 -10.74
25627 Captain, Emergency Med Svcs -9942.33

[1] 113

### Subset 2

Below code creates a subset of the mentioned columns and rows of employees getting any salary more than 0.

In [8]:
s2 <- subset(jobs, Salaries > 0, select=c("Year.Type", "Year", "Organization.Group", "Union", "Job", "Salaries"))

Below we display the top 6 rows of this subset and also the number of rows this subset has.

In [9]:
head(s2)
nrow(s2)

Year.Type Year Organization.Group            Union                         
1 Calendar  2028 General City Responsibilities Utd Pub EmpL790 SEIU-Crt Clrks
2 Calendar  2028 Public Protection             Utd Pub EmpL790 SEIU-Crt Clrks
3 Fiscal    2028 Public Protection             Utd Pub EmpL790 SEIU-Crt Clrks
4 Calendar  2019 General City Responsibilities POA                           
5 Calendar  2019 Public Protection             Municipal Attorneys Assoc     
6 Calendar  2019 General City Responsibilities Auto Machinist, Local 1414    
  Job                       Salaries 
1 Deputy Court Clerk II        674.28
2 Deputy Court Clerk II        674.28
3 Deputy Court Clerk II        674.28
4 Police Officer 3          124709.01
5 Attorney (Civil/Criminal) 155489.01
6 Automotive Machinist       69490.84

[1] 713717

### Subset 3

Below code creates a subset of the mentioned columns and rows of employees getting overtime more than 0.

In [10]:
s3 <- subset(jobs, Overtime > 0, select=c("Year.Type", "Year", "Organization.Group", "Union", "Job", "Salaries", "Overtime", "Total.Salary"))

Below displayed are the top 6 rows of this subset and also the number of rows in this subset.

In [11]:
head(s3)
nrow(s3)

Year.Type Year Organization.Group            Union                       
4  Calendar  2019 General City Responsibilities POA                         
6  Calendar  2019 General City Responsibilities Auto Machinist, Local 1414  
7  Calendar  2019 General City Responsibilities SEIU, Local 1021, Misc      
8  Calendar  2019 General City Responsibilities TWU, Local 250-A, TransitOpr
9  Calendar  2019 Community Health              Teamsters, Local 856, Spv RN
10 Calendar  2019 Public Protection             POA                         
   Job                    Salaries  Overtime  Total.Salary
4  Police Officer 3       124709.01 100499.62 230710.41   
6  Automotive Machinist    69490.84  34969.05 117804.42   
7  Public SafetyComm Disp  57062.72   6033.18  64287.90   
8  Transit Operator        74231.85  22440.63 100291.97   
9  Nursing Supervisor      22348.10   5032.61  29136.54   
10 Lieutenant 3                0.00   2911.35   2911.35

[1] 315769

### Subset 4

Below code creates a subset of the mentioned columns and rows of employees getting any other salary more than 0.

In [12]:
s4 <- subset(jobs, Other.Salaries > 0, select=c("Year.Type", "Year", "Organization.Group", "Union", "Job", "Salaries", "Other.Salaries"))

Below the top 6 rows of this subset are displayed and also the number of rows in this subset.

In [13]:
head(s4)
nrow(s4)

Year.Type Year Organization.Group            Union                         
1 Calendar  2028 General City Responsibilities Utd Pub EmpL790 SEIU-Crt Clrks
2 Calendar  2028 Public Protection             Utd Pub EmpL790 SEIU-Crt Clrks
3 Fiscal    2028 Public Protection             Utd Pub EmpL790 SEIU-Crt Clrks
4 Calendar  2019 General City Responsibilities POA                           
5 Calendar  2019 Public Protection             Municipal Attorneys Assoc     
6 Calendar  2019 General City Responsibilities Auto Machinist, Local 1414    
  Job                       Salaries  Other.Salaries
1 Deputy Court Clerk II        674.28     5.76      
2 Deputy Court Clerk II        674.28     5.76      
3 Deputy Court Clerk II        674.28     5.76      
4 Police Officer 3          124709.01  5501.78      
5 Attorney (Civil/Criminal) 155489.01  1500.00      
6 Automotive Machinist       69490.84 13344.53

[1] 478895

### Subset 5

Below code creates a subset of the mentioned columns and rows of employees getting any amount of salary after retirement. 

In [14]:
s5 <- subset(jobs, Retirement > 0, select=c("Year.Type", "Year", "Organization.Group", "Department.Code", "Union", "Job", "Salaries", "Other.Salaries", "Retirement" ))

Below the top 6 rows of this subset are displayed and also the number of rows in this subset.

In [15]:
head(s4)
nrow(s4)

Year.Type Year Organization.Group            Union                         
1 Calendar  2028 General City Responsibilities Utd Pub EmpL790 SEIU-Crt Clrks
2 Calendar  2028 Public Protection             Utd Pub EmpL790 SEIU-Crt Clrks
3 Fiscal    2028 Public Protection             Utd Pub EmpL790 SEIU-Crt Clrks
4 Calendar  2019 General City Responsibilities POA                           
5 Calendar  2019 Public Protection             Municipal Attorneys Assoc     
6 Calendar  2019 General City Responsibilities Auto Machinist, Local 1414    
  Job                       Salaries  Other.Salaries
1 Deputy Court Clerk II        674.28     5.76      
2 Deputy Court Clerk II        674.28     5.76      
3 Deputy Court Clerk II        674.28     5.76      
4 Police Officer 3          124709.01  5501.78      
5 Attorney (Civil/Criminal) 155489.01  1500.00      
6 Automotive Machinist       69490.84 13344.53

[1] 478895

### Subset 6

Below code creates a subset of the mentioned columns and rows of employees having their health and dental taken care off. 

In [16]:
s6 <- subset(jobs, Health.and.Dental > 0, select=c("Year.Type", "Year", "Organization.Group", "Union", "Job", "Salaries", "Health.and.Dental"))

Below the top 6 rows of this subset are displayed and also the number of rows in this subset.

In [17]:
head(s4)
nrow(s4)

Year.Type Year Organization.Group            Union                         
1 Calendar  2028 General City Responsibilities Utd Pub EmpL790 SEIU-Crt Clrks
2 Calendar  2028 Public Protection             Utd Pub EmpL790 SEIU-Crt Clrks
3 Fiscal    2028 Public Protection             Utd Pub EmpL790 SEIU-Crt Clrks
4 Calendar  2019 General City Responsibilities POA                           
5 Calendar  2019 Public Protection             Municipal Attorneys Assoc     
6 Calendar  2019 General City Responsibilities Auto Machinist, Local 1414    
  Job                       Salaries  Other.Salaries
1 Deputy Court Clerk II        674.28     5.76      
2 Deputy Court Clerk II        674.28     5.76      
3 Deputy Court Clerk II        674.28     5.76      
4 Police Officer 3          124709.01  5501.78      
5 Attorney (Civil/Criminal) 155489.01  1500.00      
6 Automotive Machinist       69490.84 13344.53

[1] 478895

## 4. Vectors

Here we store in one vector top 6 unique job titles from subset 1 that contains job salaries less than 0.

In [4]:
v1 <- head(unique(s1$Job))

In [5]:
v1

[1] "Captain, Emergency Med Svcs"   "Special Nurse"                
[3] "Truck Driver"                  "Firefighter"                  
[5] "General Laborer"               "Transit Car Cleaner Asst Sprv"

## 5. List

### List 1

Below we have created a list that stores total compensation that has value less than 10000.

In [13]:
l1 <- list("select count(\"Organization.Group\") from jobs where \"Organization.Group\" = 'Public Protection'")

Displaying this vector with such a large number of values slows down the kernel. 

### List 2

Below we have created a list that stores total compensation that has value between 10000 and 20000.

In [ ]:
l2 <- list(jobs$Total.Compensation > 10000 | jobs$Total.Compensation <= 20000 )

Displaying this vector with such a large number of values slows down the kernel. 

### List 3

Below we have created a list that stores total compensation that has value between 20000 and 50000.

In [ ]:
l3 <- list(jobs$Total.Compensation > 20000 | jobs$Total.Compensation <= 50000 )

Displaying this vector with such a large number of values slows down the kernel. 

### List 4

Below we have created a list that stores total compensation that has value between 50000 and 100000.

In [ ]:
l4 <- list(jobs$Total.Compensation > 50000 | jobs$Total.Compensation <= 100000 )

Displaying this vector with such a large number of values slows down the kernel. 

### List 5

Below we have created a list that stores total compensation that has value greater than 100000.

In [ ]:
l5 <- list(jobs$Total.Compensation > 100000)

Displaying this vector with such a large number of values slows down the kernel. 

## 6. Conditionals


### Conditional 1

Displayed below are the top 6 rows of Year, Organization group and job of employees having salaries less than 500 

In [16]:
head(jobs[which(jobs$Salaries < 500), c("Year", "Organization.Group", "Job", "Salaries") ])

Year Organization.Group                       Job                        
10  2019 Public Protection                        Lieutenant 3               
38  2019 Public Protection                        Police Officer 3           
195 2019 Public Protection                        Captain, Emergency Med Svcs
212 2019 Community Health                         Health Care Analyst        
213 2019 Human Welfare & Neighborhood Development Public Service Trainee     
234 2019 Public Works, Transportation & Commerce  Sergeant 3                 
    Salaries
10      0.00
38      0.00
195 -9942.33
212     0.00
213     0.00
234     0.00

### Conditional 2

Here the condition is of jobs having salaries greater than 50000.

In [17]:
head(jobs[which(jobs$Salaries > 50000), c("Year", "Organization.Group", "Job", "Salaries") ])

Year Organization.Group            Job                        Salaries 
4  2019 General City Responsibilities Police Officer 3           124709.01
5  2019 Public Protection             Attorney (Civil/Criminal)  155489.01
6  2019 General City Responsibilities Automotive Machinist        69490.84
7  2019 General City Responsibilities Public SafetyComm Disp      57062.72
8  2019 General City Responsibilities Transit Operator            74231.85
11 2019 General City Responsibilities Protective Services Worker 105677.00

### Conditional 3

Here the condition says diplay the mentioned columns and rows having overtime greater than 50000.

In [18]:
head(jobs[which(jobs$Overtime > 50000), c("Year", "Organization.Group", "Job", "Overtime", "Salaries") ])

Year Organization.Group            Job                          Overtime 
4   2019 General City Responsibilities Police Officer 3             100499.62
18  2019 Public Protection             Police Officer 2              64423.70
41  2019 General City Responsibilities Police Officer 2              65067.93
64  2019 Public Protection             Incident Support Specialist   59415.53
67  2019 General City Responsibilities Police Officer 3              81451.69
161 2019 General City Responsibilities Lieutenant, Fire Suppression  66468.76
    Salaries 
4   124709.01
18  120802.86
41  119885.26
64   94010.25
67  124685.15
161 120815.03

## 7. Loops

### Loop 1

Here we get the dimensions of out dataset in rows by columns, this dataset has 835307 rows and 22 columns.

In [ ]:
 for (i in 1:dim(jobs)[1]) {
}

In [ ]:
dim(jobs)

### Loop 2

We create a subset which we then use for looping.

In [13]:
ss1 <- subset(jobs, select=c("Year", "Job", "Salaries", "Total.Compensation"))

In [14]:
head(ss1, n=10)

Year Job                       Salaries  Total.Compensation
1  2028 Deputy Court Clerk II        674.28    864.81         
2  2028 Deputy Court Clerk II        674.28    864.81         
3  2028 Deputy Court Clerk II        674.28    864.81         
4  2019 Police Officer 3          124709.01 286685.97         
5  2019 Attorney (Civil/Criminal) 155489.01 226315.84         
6  2019 Automotive Machinist       69490.84 166377.84         
7  2019 Public SafetyComm Disp     57062.72  97952.33         
8  2019 Transit Operator           74231.85 152525.08         
9  2019 Nursing Supervisor         22348.10  35363.75         
10 2019 Lieutenant 3                   0.00   2911.35

The below loop is for the values in the Total compensation column being less than or equal to 1000.

In [ ]:
sum <- 0
for (i in 1:nrow(ss1)){
    if(ss1$Total.Compensation[i] <= 1000){
        sum <- sum + 1
    }
   
} 

### Loop 3

The below loop is for the values in the Salaries column being less than or equal to 100.

In [ ]:
sum <- 0
for (i in 1:nrow(ss1)){
    if(ss1$Salaries[i] <= 100){
        sum <- sum + 1
    }
   
} 

## 8. Summary

Looking at the above analysis we learn that there are a few groups of departments that get less than normal salary and some that do not get paid on time. Also the retirment plan, health and dental insurance amount are not equal for all and some do not even get these benefits. The margin between the different job types and the salaries or even within a department is large.   

## 9. Recommendations

After analysis we can say that the income in the city does not do justice to certain deparments of job types. There need to be some standards and rules in place that do not deprive any employee of the salary and benefits they deserve. The San Francisco Chronicle newspaper should highlight the departments that do not get paid well and also discuss the overtime wages, retirement plan and the health insurance benefits that employees deserver but not all receive. 